### ISSUE TITLE
Move MailTarget into NLog.MailTarget package

### ISSUE THREAD

**Issue by snakefoot** [NLog/NLog#2786]
Remove the following dependencies from NLog-core:

```
<Reference Include="System.Net" />
<Reference Include="System.Net.Mail" />
```

Probably for NLog ver. 6.0 (Be careful not to step on the toes of NLog.MailKit - https://www.nuget.org/packages/NLog.MailKit/)

**Comment by 304NotModified** [comment]
proposal: NLog.MailTarget

it's not only smtp ;)

**Comment by snakefoot** [comment]
> @304NotModified  proposal: NLog.MailTarget

Updated the title

**Comment by 304NotModified** [comment]
> (Be careful not to step on the toes of NLog.MailKit - https://www.nuget.org/packages/NLog.MailKit/)

Could you elaborate on this? Do you mean, don't call it NLog.MailKit?

**Comment by snakefoot** [comment]
> Could you elaborate on this?

Just curious how they would live side-by-side, when both are called "mail"-target

**Comment by 304NotModified** [comment]
good one. Including both will only keep one, the last included. 

That could be tricky with auto loading.

**Comment by 304NotModified** [comment]
Must have also because of https://github.com/NLog/NLog/pull/3978 and #1351

**Comment by snakefoot** [comment]
Resolved by #5635 for NLog v6
